vault/testing.go

```diff
@@ -1366,6 +1367,8 @@ func (tl *TestLogger) StopLogging() {
 // given, one will be generated based on t.Name() for the cluster logger, and if
 // no base.Logger is given will also be used as the basis for each core's logger.
 func NewTestCluster(t testing.T, base *CoreConfig, opts *TestClusterOptions) *TestCluster {
+	base.DetectDeadlocks = TestDeadlockDetection
```

Comment:
Have you verified this is sufficient?  I would think that we'd also need to assign to coreConfig further down.  In addition, there are two problems I see with this approach.  One, `base` can be nil, which would make this panic.  Two, it's rude to modify an input parameter, unless that's the func's stated purpose.